src/proto.js

```diff
@@ -182,6 +184,65 @@ exports.unlock = async function (key, fn) {
   }
 }
 
+/**
+ * Fake superagent request that doesn't actually send anything
+ * but just persists the request in memory for tracking.
+ * Later, for "chatty" destinations it can use control requests
+ * as its data source to give some responses.
+ */
+function mockRequest (method, url, onEnd) {
+  return {
+    method,
+    url,
+    onEnd,
+    headers: {
+      // Default headers.
+      'User-Agent': 'Segment.io/1.0',
+      'Accept-Encoding': 'identity',
+      'Content-Type': 'application/json'
+    },
+    // Current subset of functions here represents just what is necessary
+    // to make webhooks work.
+    post (url) {
+      this.url = url
+      return this
+    },
+    get (url) {
+      this.url = url
+      return this
+    },
+    redirects (redirects) {
+      this.redirects = redirects
+      return this
+    },
+    type (type) {
+      this.type = type
+      return this
+    },
+    send (body) {
+      this.body = body
+      return this
+    },
+    parse (ignore) {
+      return this
+    },
+    set (header, value) {
+      this.headers[header] = value
+      return this
+    },
+    query () {
+      return
```

Comment:
This should be `return this` 

Probably just better to remove it until we support it really.